runbotics-orchestrator/src/main/java/com/runbotics/web/rest/BotCollectionResource.java

```diff
@@ -154,6 +178,15 @@ public ResponseEntity<BotCollectionDTO> getBotCollection(@PathVariable UUID id)
     @DeleteMapping("bot-collection/{id}")
     public ResponseEntity<Void> deleteBotCollection(@PathVariable UUID id) {
         log.debug("REST request to delete BotCollection : {}", id);
+
+        if(isPublicOrGuest(id)){
+            log.debug("Can not delete collection with id: {}", id);
+            return ResponseEntity
+                .noContent()
+                .headers(HeaderUtil.createFailureAlert(applicationName, true, ENTITY_NAME, HttpStatus.FORBIDDEN.toString(), "Can't delete collection"))
+                .build();
+        }
```

Comment:
create exception class and throw an exception instead of using responseEntity in business logic